# بسم الله الرحمن الرحيم

In [48]:
from utils import *
import pandas as pd 
from sklearn.model_selection import train_test_split


## Reading the Dataset, cleaning , preparing it for the training 

In [49]:
df = pd.read_csv("dataset/body_level_classification_train.csv")

In [50]:
df_original= cleaning_data(df).copy()

In [51]:
all_class_train,all_class_test=data_spliting(df_original,1/5)

In [52]:
oversampled= over_sampling(all_class_train,sampling_ratio=0.8)

In [53]:
X_train=oversampled.drop(["Body_Level"],axis=1).to_numpy()
Y_train=oversampled["Body_Level"].to_numpy()
X_test =all_class_test.drop(["Body_Level"],axis=1).to_numpy()
Y_test =all_class_test["Body_Level"].to_numpy()

In [54]:
X_train, scaler= data_scaling(X_train)
with open(f"scaler","wb") as fp : 
    pickle.dump(scaler,fp)

In [55]:
X_val, X_test, Y_val, Y_test = train_test_split(X_test
                                                     , Y_test, test_size=0.5, random_state=42)

In [56]:
X_val = scaler.transform(X_val)
X_test=scaler.transform(X_test)

In [57]:
model=build_model(2,0.01)

In [58]:
history = model.fit(X_train,Y_train,validation_data=[X_val,Y_val] 
             ,batch_size=32, epochs=400 , verbose =1, 
                       callbacks=[CustomCallback("NNModels/after_sampling/fifth_model.h5")])

Epoch 1/400
46/46 [==============================] - 2s 10ms/step - loss: 0.9504 - accuracy: 0.4565 - val_loss: 0.4494 - val_accuracy: 0.6531
Epoch 2/400
46/46 [==============================] - 0s 4ms/step - loss: 0.3774 - accuracy: 0.6531 - val_loss: 0.2885 - val_accuracy: 0.7415
Epoch 3/400
46/46 [==============================] - 0s 4ms/step - loss: 0.2956 - accuracy: 0.7265 - val_loss: 0.2439 - val_accuracy: 0.7823
Epoch 4/400
46/46 [==============================] - 0s 3ms/step - loss: 0.2602 - accuracy: 0.7714 - val_loss: 0.2137 - val_accuracy: 0.8503
Epoch 5/400
46/46 [==============================] - 0s 3ms/step - loss: 0.2367 - accuracy: 0.8163 - val_loss: 0.2008 - val_accuracy: 0.8367
Epoch 6/400
46/46 [==============================] - 0s 3ms/step - loss: 0.2176 - accuracy: 0.8156 - val_loss: 0.1819 - val_accuracy: 0.8503
Epoch 7/400
46/46 [==============================] - 0s 3ms/step - loss: 0.2030 - accuracy: 0.8231 - val_loss: 0.1711 - val_accuracy: 0.8707
Epoch 8/400


metadata.json                                  2023-05-16 21:20:34           64
46/46 [==============================] - 0s 3ms/step - loss: 0.0539 - accuracy: 0.9639 - val_loss: 0.0545 - val_accuracy: 0.9728
Epoch 49/400
46/46 [==============================] - 0s 3ms/step - loss: 0.0524 - accuracy: 0.9565 - val_loss: 0.0551 - val_accuracy: 0.9796
Epoch 50/400
46/46 [==============================] - 0s 3ms/step - loss: 0.0522 - accuracy: 0.9592 - val_loss: 0.0543 - val_accuracy: 0.9796
Epoch 51/400
46/46 [==============================] - 0s 2ms/step - loss: 0.0512 - accuracy: 0.9633 - val_loss: 0.0528 - val_accuracy: 0.9796
Epoch 52/400
46/46 [==============================] - 0s 2ms/step - loss: 0.0499 - accuracy: 0.9605 - val_loss: 0.0507 - val_accuracy: 0.9796
Epoch 53/400
46/46 [==============================] - 0s 3ms/step - loss: 0.0497 - accuracy: 0.9646 - val_loss: 0.0530 - val_accuracy: 0.9796
Epoch 54/400
46/46 [==============================] - 0s 3ms/step - loss: 0.0487 

File Name                                             Modified             Size
variables.h5                                   2023-05-16 21:20:37        17192
config.json                                    2023-05-16 21:20:37         1588
metadata.json                                  2023-05-16 21:20:37           64
46/46 [==============================] - 0s 3ms/step - loss: 0.0367 - accuracy: 0.9714 - val_loss: 0.0453 - val_accuracy: 0.9796
Epoch 74/400
28/46 [=================>............] - ETA: 0s - loss: 0.0360 - accuracy: 0.9676
Yes You are here 0.9700680375099182 0.9795918464660645
Keras weights file (<HDF5 file "variables.h5" (mode r+)>) saving:
...layers
......dense
.........vars
............0
............1
...metrics
......mean
.........vars
............0
............1
......mean_metric_wrapper
.........vars
............0
............1
...optimizer
......vars
.........0
.........1
.........2
.........3
.........4
...vars
Keras model archive saving:
File Name              

config.json                                    2023-05-16 21:20:39         1588
metadata.json                                  2023-05-16 21:20:39           64
46/46 [==============================] - 0s 3ms/step - loss: 0.0335 - accuracy: 0.9721 - val_loss: 0.0390 - val_accuracy: 0.9796
Epoch 84/400
24/46 [==============>...............] - ETA: 0s - loss: 0.0322 - accuracy: 0.9740
Yes You are here 0.9734693765640259 0.9795918464660645
Keras weights file (<HDF5 file "variables.h5" (mode r+)>) saving:
...layers
......dense
.........vars
............0
............1
...metrics
......mean
.........vars
............0
............1
......mean_metric_wrapper
.........vars
............0
............1
...optimizer
......vars
.........0
.........1
.........2
.........3
.........4
...vars
Keras model archive saving:
File Name                                             Modified             Size
variables.h5                                   2023-05-16 21:20:39        17192
config.json            

29/46 [=================>............] - ETA: 0s - loss: 0.0273 - accuracy: 0.9849
Yes You are here 0.9775510430335999 0.9795918464660645
Keras weights file (<HDF5 file "variables.h5" (mode r+)>) saving:
...layers
......dense
.........vars
............0
............1
...metrics
......mean
.........vars
............0
............1
......mean_metric_wrapper
.........vars
............0
............1
...optimizer
......vars
.........0
.........1
.........2
.........3
.........4
...vars
Keras model archive saving:
File Name                                             Modified             Size
variables.h5                                   2023-05-16 21:20:40        17192
config.json                                    2023-05-16 21:20:40         1588
metadata.json                                  2023-05-16 21:20:40           64
46/46 [==============================] - 0s 3ms/step - loss: 0.0295 - accuracy: 0.9776 - val_loss: 0.0441 - val_accuracy: 0.9796
Epoch 93/400
46/46 [================

File Name                                             Modified             Size
variables.h5                                   2023-05-16 21:20:42        17192
config.json                                    2023-05-16 21:20:42         1588
metadata.json                                  2023-05-16 21:20:42           64
46/46 [==============================] - 0s 3ms/step - loss: 0.0269 - accuracy: 0.9776 - val_loss: 0.0383 - val_accuracy: 0.9796
Epoch 102/400
30/46 [==================>...........] - ETA: 0s - loss: 0.0267 - accuracy: 0.9792
Yes You are here 0.9768707752227783 0.9795918464660645
Keras weights file (<HDF5 file "variables.h5" (mode r+)>) saving:
...layers
......dense
.........vars
............0
............1
...metrics
......mean
.........vars
............0
............1
......mean_metric_wrapper
.........vars
............0
............1
...optimizer
......vars
.........0
.........1
.........2
.........3
.........4
...vars
Keras model archive saving:
File Name             

metadata.json                                  2023-05-16 21:20:43           64
46/46 [==============================] - 0s 3ms/step - loss: 0.0252 - accuracy: 0.9748 - val_loss: 0.0399 - val_accuracy: 0.9796
Epoch 111/400
25/46 [===============>..............] - ETA: 0s - loss: 0.0274 - accuracy: 0.9725
Yes You are here 0.9775510430335999 0.9795918464660645
Keras weights file (<HDF5 file "variables.h5" (mode r+)>) saving:
...layers
......dense
.........vars
............0
............1
...metrics
......mean
.........vars
............0
............1
......mean_metric_wrapper
.........vars
............0
............1
...optimizer
......vars
.........0
.........1
.........2
.........3
.........4
...vars
Keras model archive saving:
File Name                                             Modified             Size
variables.h5                                   2023-05-16 21:20:43        17192
config.json                                    2023-05-16 21:20:43         1588
metadata.json         

25/46 [===============>..............] - ETA: 0s - loss: 0.0231 - accuracy: 0.9837
Yes You are here 0.981632649898529 0.9795918464660645
Keras weights file (<HDF5 file "variables.h5" (mode r+)>) saving:
...layers
......dense
.........vars
............0
............1
...metrics
......mean
.........vars
............0
............1
......mean_metric_wrapper
.........vars
............0
............1
...optimizer
......vars
.........0
.........1
.........2
.........3
.........4
...vars
Keras model archive saving:
File Name                                             Modified             Size
variables.h5                                   2023-05-16 21:20:44        17192
config.json                                    2023-05-16 21:20:44         1588
metadata.json                                  2023-05-16 21:20:44           64
46/46 [==============================] - 0s 4ms/step - loss: 0.0234 - accuracy: 0.9816 - val_loss: 0.0382 - val_accuracy: 0.9796
Epoch 121/400
46/46 [================

File Name                                             Modified             Size
variables.h5                                   2023-05-16 21:20:46        17192
config.json                                    2023-05-16 21:20:46         1588
metadata.json                                  2023-05-16 21:20:46           64
46/46 [==============================] - 0s 3ms/step - loss: 0.0240 - accuracy: 0.9776 - val_loss: 0.0453 - val_accuracy: 0.9864
Epoch 131/400
31/46 [===================>..........] - ETA: 0s - loss: 0.0208 - accuracy: 0.9839
Yes You are here 0.9823129177093506 0.9863945841789246
Keras weights file (<HDF5 file "variables.h5" (mode r+)>) saving:
...layers
......dense
.........vars
............0
............1
...metrics
......mean
.........vars
............0
............1
......mean_metric_wrapper
.........vars
............0
............1
...optimizer
......vars
.........0
.........1
.........2
.........3
.........4
...vars
Keras model archive saving:
File Name             

Epoch 151/400
46/46 [==============================] - 0s 3ms/step - loss: 0.0182 - accuracy: 0.9850 - val_loss: 0.0413 - val_accuracy: 0.9796
Epoch 152/400
46/46 [==============================] - 0s 2ms/step - loss: 0.0183 - accuracy: 0.9844 - val_loss: 0.0390 - val_accuracy: 0.9796
Epoch 153/400
46/46 [==============================] - 0s 2ms/step - loss: 0.0179 - accuracy: 0.9864 - val_loss: 0.0393 - val_accuracy: 0.9796
Epoch 154/400
28/46 [=================>............] - ETA: 0s - loss: 0.0176 - accuracy: 0.9810
Yes You are here 0.9809523820877075 0.9863945841789246
Keras weights file (<HDF5 file "variables.h5" (mode r+)>) saving:
...layers
......dense
.........vars
............0
............1
...metrics
......mean
.........vars
............0
............1
......mean_metric_wrapper
.........vars
............0
............1
...optimizer
......vars
.........0
.........1
.........2
.........3
.........4
...vars
Keras model archive saving:
File Name                                 

28/46 [=================>............] - ETA: 0s - loss: 0.0231 - accuracy: 0.9777
Yes You are here 0.980272114276886 0.9863945841789246
Keras weights file (<HDF5 file "variables.h5" (mode r+)>) saving:
...layers
......dense
.........vars
............0
............1
...metrics
......mean
.........vars
............0
............1
......mean_metric_wrapper
.........vars
............0
............1
...optimizer
......vars
.........0
.........1
.........2
.........3
.........4
...vars
Keras model archive saving:
File Name                                             Modified             Size
variables.h5                                   2023-05-16 21:20:51        17192
config.json                                    2023-05-16 21:20:51         1588
metadata.json                                  2023-05-16 21:20:51           64
46/46 [==============================] - 0s 3ms/step - loss: 0.0219 - accuracy: 0.9803 - val_loss: 0.0381 - val_accuracy: 0.9864
Epoch 168/400
29/46 [================

46/46 [==============================] - 0s 3ms/step - loss: 0.0159 - accuracy: 0.9864 - val_loss: 0.0407 - val_accuracy: 0.9864
Epoch 184/400
46/46 [==============================] - 0s 3ms/step - loss: 0.0151 - accuracy: 0.9871 - val_loss: 0.0423 - val_accuracy: 0.9796
Epoch 185/400
46/46 [==============================] - 0s 2ms/step - loss: 0.0157 - accuracy: 0.9857 - val_loss: 0.0405 - val_accuracy: 0.9796
Epoch 186/400
31/46 [===================>..........] - ETA: 0s - loss: 0.0159 - accuracy: 0.9839
Yes You are here 0.9857142567634583 0.9863945841789246
Keras weights file (<HDF5 file "variables.h5" (mode r+)>) saving:
...layers
......dense
.........vars
............0
............1
...metrics
......mean
.........vars
............0
............1
......mean_metric_wrapper
.........vars
............0
............1
...optimizer
......vars
.........0
.........1
.........2
.........3
.........4
...vars
Keras model archive saving:
File Name                                             Mo

46/46 [==============================] - 0s 3ms/step - loss: 0.0148 - accuracy: 0.9857 - val_loss: 0.0404 - val_accuracy: 0.9796
Epoch 200/400
46/46 [==============================] - 0s 3ms/step - loss: 0.0146 - accuracy: 0.9844 - val_loss: 0.0347 - val_accuracy: 0.9796
Epoch 201/400
46/46 [==============================] - 0s 3ms/step - loss: 0.0146 - accuracy: 0.9871 - val_loss: 0.0401 - val_accuracy: 0.9796
Epoch 202/400
24/46 [==============>...............] - ETA: 0s - loss: 0.0135 - accuracy: 0.9909
Yes You are here 0.9850339889526367 0.9863945841789246
Keras weights file (<HDF5 file "variables.h5" (mode r+)>) saving:
...layers
......dense
.........vars
............0
............1
...metrics
......mean
.........vars
............0
............1
......mean_metric_wrapper
.........vars
............0
............1
...optimizer
......vars
.........0
.........1
.........2
.........3
.........4
...vars
Keras model archive saving:
File Name                                             Mo

metadata.json                                  2023-05-16 21:20:57           64
46/46 [==============================] - 0s 4ms/step - loss: 0.0133 - accuracy: 0.9905 - val_loss: 0.0373 - val_accuracy: 0.9864
Epoch 217/400
46/46 [==============================] - 0s 3ms/step - loss: 0.0142 - accuracy: 0.9850 - val_loss: 0.0337 - val_accuracy: 0.9796
Epoch 218/400
31/46 [===================>..........] - ETA: 0s - loss: 0.0129 - accuracy: 0.9849
Yes You are here 0.9857142567634583 0.9863945841789246
Keras weights file (<HDF5 file "variables.h5" (mode r+)>) saving:
...layers
......dense
.........vars
............0
............1
...metrics
......mean
.........vars
............0
............1
......mean_metric_wrapper
.........vars
............0
............1
...optimizer
......vars
.........0
.........1
.........2
.........3
.........4
...vars
Keras model archive saving:
File Name                                             Modified             Size
variables.h5                           

File Name                                             Modified             Size
variables.h5                                   2023-05-16 21:20:59        17192
config.json                                    2023-05-16 21:20:59         1588
metadata.json                                  2023-05-16 21:20:59           64
46/46 [==============================] - 0s 4ms/step - loss: 0.0131 - accuracy: 0.9878 - val_loss: 0.0403 - val_accuracy: 0.9864
Epoch 230/400
46/46 [==============================] - 0s 3ms/step - loss: 0.0130 - accuracy: 0.9891 - val_loss: 0.0403 - val_accuracy: 0.9796
Epoch 231/400
29/46 [=================>............] - ETA: 0s - loss: 0.0120 - accuracy: 0.9871
Yes You are here 0.9823129177093506 0.9863945841789246
Keras weights file (<HDF5 file "variables.h5" (mode r+)>) saving:
...layers
......dense
.........vars
............0
............1
...metrics
......mean
.........vars
............0
............1
......mean_metric_wrapper
.........vars
............0
........

File Name                                             Modified             Size
variables.h5                                   2023-05-16 21:21:01        17192
config.json                                    2023-05-16 21:21:01         1588
metadata.json                                  2023-05-16 21:21:01           64
46/46 [==============================] - 0s 3ms/step - loss: 0.0135 - accuracy: 0.9837 - val_loss: 0.0390 - val_accuracy: 0.9864
Epoch 242/400
46/46 [==============================] - 0s 3ms/step - loss: 0.0133 - accuracy: 0.9864 - val_loss: 0.0393 - val_accuracy: 0.9796
Epoch 243/400
46/46 [==============================] - 0s 3ms/step - loss: 0.0123 - accuracy: 0.9884 - val_loss: 0.0387 - val_accuracy: 0.9796
Epoch 244/400
46/46 [==============================] - 0s 3ms/step - loss: 0.0120 - accuracy: 0.9884 - val_loss: 0.0358 - val_accuracy: 0.9796
Epoch 245/400
46/46 [==============================] - 0s 3ms/step - loss: 0.0126 - accuracy: 0.9878 - val_loss: 0.0340 - 

File Name                                             Modified             Size
variables.h5                                   2023-05-16 21:21:03        17192
config.json                                    2023-05-16 21:21:03         1588
metadata.json                                  2023-05-16 21:21:03           64
46/46 [==============================] - 0s 4ms/step - loss: 0.0116 - accuracy: 0.9898 - val_loss: 0.0328 - val_accuracy: 0.9864
Epoch 255/400
24/46 [==============>...............] - ETA: 0s - loss: 0.0131 - accuracy: 0.9844
Yes You are here 0.9857142567634583 0.9863945841789246
Keras weights file (<HDF5 file "variables.h5" (mode r+)>) saving:
...layers
......dense
.........vars
............0
............1
...metrics
......mean
.........vars
............0
............1
......mean_metric_wrapper
.........vars
............0
............1
...optimizer
......vars
.........0
.........1
.........2
.........3
.........4
...vars
Keras model archive saving:
File Name             

File Name                                             Modified             Size
variables.h5                                   2023-05-16 21:21:05        17192
config.json                                    2023-05-16 21:21:05         1588
metadata.json                                  2023-05-16 21:21:05           64
46/46 [==============================] - 0s 4ms/step - loss: 0.0113 - accuracy: 0.9912 - val_loss: 0.0363 - val_accuracy: 0.9864
Epoch 268/400
29/46 [=================>............] - ETA: 0s - loss: 0.0116 - accuracy: 0.9881
Yes You are here 0.9891156554222107 0.9863945841789246
Keras weights file (<HDF5 file "variables.h5" (mode r+)>) saving:
...layers
......dense
.........vars
............0
............1
...metrics
......mean
.........vars
............0
............1
......mean_metric_wrapper
.........vars
............0
............1
...optimizer
......vars
.........0
.........1
.........2
.........3
.........4
...vars
Keras model archive saving:
File Name             

File Name                                             Modified             Size
variables.h5                                   2023-05-16 21:21:07        17192
config.json                                    2023-05-16 21:21:07         1588
metadata.json                                  2023-05-16 21:21:07           64
46/46 [==============================] - 0s 2ms/step - loss: 0.0119 - accuracy: 0.9878 - val_loss: 0.0331 - val_accuracy: 0.9864
Epoch 280/400
34/46 [=====================>........] - ETA: 0s - loss: 0.0105 - accuracy: 0.9917
Yes You are here 0.9904761910438538 0.9863945841789246
Keras weights file (<HDF5 file "variables.h5" (mode r+)>) saving:
...layers
......dense
.........vars
............0
............1
...metrics
......mean
.........vars
............0
............1
......mean_metric_wrapper
.........vars
............0
............1
...optimizer
......vars
.........0
.........1
.........2
.........3
.........4
...vars
Keras model archive saving:
File Name             

File Name                                             Modified             Size
variables.h5                                   2023-05-16 21:21:08        17192
config.json                                    2023-05-16 21:21:08         1588
metadata.json                                  2023-05-16 21:21:08           64
46/46 [==============================] - 0s 3ms/step - loss: 0.0111 - accuracy: 0.9884 - val_loss: 0.0359 - val_accuracy: 0.9864
Epoch 291/400
28/46 [=================>............] - ETA: 0s - loss: 0.0106 - accuracy: 0.9900
Yes You are here 0.9897959232330322 0.9863945841789246
Keras weights file (<HDF5 file "variables.h5" (mode r+)>) saving:
...layers
......dense
.........vars
............0
............1
...metrics
......mean
.........vars
............0
............1
......mean_metric_wrapper
.........vars
............0
............1
...optimizer
......vars
.........0
.........1
.........2
.........3
.........4
...vars
Keras model archive saving:
File Name             

metadata.json                                  2023-05-16 21:21:10           64
46/46 [==============================] - 0s 3ms/step - loss: 0.0145 - accuracy: 0.9844 - val_loss: 0.0368 - val_accuracy: 0.9864
Epoch 300/400
28/46 [=================>............] - ETA: 0s - loss: 0.0094 - accuracy: 0.9922
Yes You are here 0.9877551198005676 0.9863945841789246
Keras weights file (<HDF5 file "variables.h5" (mode r+)>) saving:
...layers
......dense
.........vars
............0
............1
...metrics
......mean
.........vars
............0
............1
......mean_metric_wrapper
.........vars
............0
............1
...optimizer
......vars
.........0
.........1
.........2
.........3
.........4
...vars
Keras model archive saving:
File Name                                             Modified             Size
variables.h5                                   2023-05-16 21:21:10        17192
config.json                                    2023-05-16 21:21:10         1588
metadata.json         

File Name                                             Modified             Size
variables.h5                                   2023-05-16 21:21:11        17192
config.json                                    2023-05-16 21:21:11         1588
metadata.json                                  2023-05-16 21:21:11           64
46/46 [==============================] - 0s 4ms/step - loss: 0.0104 - accuracy: 0.9878 - val_loss: 0.0363 - val_accuracy: 0.9864
Epoch 309/400
46/46 [==============================] - ETA: 0s - loss: 0.0101 - accuracy: 0.9925
Yes You are here 0.9925169944763184 0.9863945841789246
Keras weights file (<HDF5 file "variables.h5" (mode r+)>) saving:
...layers
......dense
.........vars
............0
............1
...metrics
......mean
.........vars
............0
............1
......mean_metric_wrapper
.........vars
............0
............1
...optimizer
......vars
.........0
.........1
.........2
.........3
.........4
...vars
Keras model archive saving:
File Name             

Epoch 317/400
34/46 [=====================>........] - ETA: 0s - loss: 0.0100 - accuracy: 0.9908
Yes You are here 0.9925169944763184 0.9863945841789246
Keras weights file (<HDF5 file "variables.h5" (mode r+)>) saving:
...layers
......dense
.........vars
............0
............1
...metrics
......mean
.........vars
............0
............1
......mean_metric_wrapper
.........vars
............0
............1
...optimizer
......vars
.........0
.........1
.........2
.........3
.........4
...vars
Keras model archive saving:
File Name                                             Modified             Size
variables.h5                                   2023-05-16 21:21:13        17192
config.json                                    2023-05-16 21:21:13         1588
metadata.json                                  2023-05-16 21:21:13           64
46/46 [==============================] - 0s 3ms/step - loss: 0.0101 - accuracy: 0.9925 - val_loss: 0.0296 - val_accuracy: 0.9864
Epoch 318/400
28/46 [=

File Name                                             Modified             Size
variables.h5                                   2023-05-16 21:21:14        17192
config.json                                    2023-05-16 21:21:14         1588
metadata.json                                  2023-05-16 21:21:14           64
46/46 [==============================] - 0s 4ms/step - loss: 0.0099 - accuracy: 0.9925 - val_loss: 0.0309 - val_accuracy: 0.9864
Epoch 326/400
25/46 [===============>..............] - ETA: 0s - loss: 0.0087 - accuracy: 0.9900
Yes You are here 0.9904761910438538 0.9863945841789246
Keras weights file (<HDF5 file "variables.h5" (mode r+)>) saving:
...layers
......dense
.........vars
............0
............1
...metrics
......mean
.........vars
............0
............1
......mean_metric_wrapper
.........vars
............0
............1
...optimizer
......vars
.........0
.........1
.........2
.........3
.........4
...vars
Keras model archive saving:
File Name             

metadata.json                                  2023-05-16 21:21:15           64
46/46 [==============================] - 0s 3ms/step - loss: 0.0094 - accuracy: 0.9912 - val_loss: 0.0328 - val_accuracy: 0.9864
Epoch 335/400
29/46 [=================>............] - ETA: 0s - loss: 0.0108 - accuracy: 0.9892
Yes You are here 0.9904761910438538 0.9863945841789246
Keras weights file (<HDF5 file "variables.h5" (mode r+)>) saving:
...layers
......dense
.........vars
............0
............1
...metrics
......mean
.........vars
............0
............1
......mean_metric_wrapper
.........vars
............0
............1
...optimizer
......vars
.........0
.........1
.........2
.........3
.........4
...vars
Keras model archive saving:
File Name                                             Modified             Size
variables.h5                                   2023-05-16 21:21:15        17192
config.json                                    2023-05-16 21:21:15         1588
metadata.json         

variables.h5                                   2023-05-16 21:21:17        17192
config.json                                    2023-05-16 21:21:17         1588
metadata.json                                  2023-05-16 21:21:17           64
46/46 [==============================] - 0s 3ms/step - loss: 0.0093 - accuracy: 0.9946 - val_loss: 0.0332 - val_accuracy: 0.9864
Epoch 347/400
29/46 [=================>............] - ETA: 0s - loss: 0.0077 - accuracy: 0.9957
Yes You are here 0.9911564588546753 0.9863945841789246
Keras weights file (<HDF5 file "variables.h5" (mode r+)>) saving:
...layers
......dense
.........vars
............0
............1
...metrics
......mean
.........vars
............0
............1
......mean_metric_wrapper
.........vars
............0
............1
...optimizer
......vars
.........0
.........1
.........2
.........3
.........4
...vars
Keras model archive saving:
File Name                                             Modified             Size
variables.h5          

46/46 [==============================] - 0s 3ms/step - loss: 0.0091 - accuracy: 0.9925 - val_loss: 0.0323 - val_accuracy: 0.9864
Epoch 356/400
31/46 [===================>..........] - ETA: 0s - loss: 0.0087 - accuracy: 0.9929
Yes You are here 0.9904761910438538 0.9863945841789246
Keras weights file (<HDF5 file "variables.h5" (mode r+)>) saving:
...layers
......dense
.........vars
............0
............1
...metrics
......mean
.........vars
............0
............1
......mean_metric_wrapper
.........vars
............0
............1
...optimizer
......vars
.........0
.........1
.........2
.........3
.........4
...vars
Keras model archive saving:
File Name                                             Modified             Size
variables.h5                                   2023-05-16 21:21:18        17192
config.json                                    2023-05-16 21:21:18         1588
metadata.json                                  2023-05-16 21:21:18           64
46/46 [===============

File Name                                             Modified             Size
variables.h5                                   2023-05-16 21:21:20        17192
config.json                                    2023-05-16 21:21:20         1588
metadata.json                                  2023-05-16 21:21:20           64
46/46 [==============================] - 0s 3ms/step - loss: 0.0093 - accuracy: 0.9912 - val_loss: 0.0317 - val_accuracy: 0.9864
Epoch 365/400
30/46 [==================>...........] - ETA: 0s - loss: 0.0087 - accuracy: 0.9885
Yes You are here 0.9891156554222107 0.9863945841789246
Keras weights file (<HDF5 file "variables.h5" (mode r+)>) saving:
...layers
......dense
.........vars
............0
............1
...metrics
......mean
.........vars
............0
............1
......mean_metric_wrapper
.........vars
............0
............1
...optimizer
......vars
.........0
.........1
.........2
.........3
.........4
...vars
Keras model archive saving:
File Name             

Epoch 373/400
38/46 [=======================>......] - ETA: 0s - loss: 0.0090 - accuracy: 0.9918
Yes You are here 0.9897959232330322 0.9863945841789246
Keras weights file (<HDF5 file "variables.h5" (mode r+)>) saving:
...layers
......dense
.........vars
............0
............1
...metrics
......mean
.........vars
............0
............1
......mean_metric_wrapper
.........vars
............0
............1
...optimizer
......vars
.........0
.........1
.........2
.........3
.........4
...vars
Keras model archive saving:
File Name                                             Modified             Size
variables.h5                                   2023-05-16 21:21:21        17192
config.json                                    2023-05-16 21:21:21         1588
metadata.json                                  2023-05-16 21:21:21           64
46/46 [==============================] - 0s 3ms/step - loss: 0.0095 - accuracy: 0.9898 - val_loss: 0.0283 - val_accuracy: 0.9864
Epoch 374/400
46/46 [=

File Name                                             Modified             Size
variables.h5                                   2023-05-16 21:21:22        17192
config.json                                    2023-05-16 21:21:22         1588
metadata.json                                  2023-05-16 21:21:22           64
46/46 [==============================] - 0s 3ms/step - loss: 0.0090 - accuracy: 0.9925 - val_loss: 0.0288 - val_accuracy: 0.9864
Epoch 383/400
30/46 [==================>...........] - ETA: 0s - loss: 0.0078 - accuracy: 0.9937
Yes You are here 0.9925169944763184 0.9863945841789246
Keras weights file (<HDF5 file "variables.h5" (mode r+)>) saving:
...layers
......dense
.........vars
............0
............1
...metrics
......mean
.........vars
............0
............1
......mean_metric_wrapper
.........vars
............0
............1
...optimizer
......vars
.........0
.........1
.........2
.........3
.........4
...vars
Keras model archive saving:
File Name             

metadata.json                                  2023-05-16 21:21:23           64
46/46 [==============================] - 0s 3ms/step - loss: 0.0090 - accuracy: 0.9898 - val_loss: 0.0320 - val_accuracy: 0.9864
Epoch 392/400
46/46 [==============================] - 0s 3ms/step - loss: 0.0094 - accuracy: 0.9912 - val_loss: 0.0326 - val_accuracy: 0.9796
Epoch 393/400
31/46 [===================>..........] - ETA: 0s - loss: 0.0090 - accuracy: 0.9899
Yes You are here 0.9897959232330322 0.9863945841789246
Keras weights file (<HDF5 file "variables.h5" (mode r+)>) saving:
...layers
......dense
.........vars
............0
............1
...metrics
......mean
.........vars
............0
............1
......mean_metric_wrapper
.........vars
............0
............1
...optimizer
......vars
.........0
.........1
.........2
.........3
.........4
...vars
Keras model archive saving:
File Name                                             Modified             Size
variables.h5                           

In [59]:
accurate_train_accuracy,classify_report= test_model(model,X_train,Y_train)

46/46 [==============================] - 0s 1ms/step


In [60]:
accurate_train_accuracy

0.9891156462585035

In [61]:
print(classify_report)

              precision    recall  f1-score   support

           0       0.96      1.00      0.98       272
           1       0.99      0.95      0.97       278
           2       0.99      0.99      0.99       376
           3       1.00      1.00      1.00       544

    accuracy                           0.99      1470
   macro avg       0.99      0.99      0.99      1470
weighted avg       0.99      0.99      0.99      1470



In [62]:
accurate_val_accuracy,classify_report= test_model(model,X_val,Y_val)

5/5 [==============================] - 0s 1ms/step


In [63]:
accurate_val_accuracy

0.9863945578231292

In [64]:
print(classify_report)

              precision    recall  f1-score   support

           0       1.00      0.95      0.97        20
           1       0.95      1.00      0.98        20
           2       0.97      1.00      0.99        36
           3       1.00      0.99      0.99        71

    accuracy                           0.99       147
   macro avg       0.98      0.98      0.98       147
weighted avg       0.99      0.99      0.99       147



In [65]:
accurate_test_accuracy,classify_report= test_model(model,X_test,Y_test)

5/5 [==============================] - 0s 2ms/step


In [66]:
accurate_test_accuracy

0.9932432432432432

In [20]:
print(classify_report)

              precision    recall  f1-score   support

           0       1.00      1.00      1.00        18
           1       1.00      0.95      0.97        20
           2       0.98      0.98      0.98        45
           3       0.98      1.00      0.99        65

    accuracy                           0.99       148
   macro avg       0.99      0.98      0.99       148
weighted avg       0.99      0.99      0.99       148



## Notes

- name            : first model 
- split           : 1/3 
- oversampling    : 0.5 
- val_acc         : 0.9836734693877551
- test_acc        : 0.983739837398374
- gamma           : 1 
---------------------------------------------------------------------------------------------
- name            : second model 
- split           : 1/3 
- oversampling    : 0.8
- val_acc         : 0.9877551020408163
- test_acc        : 0.9878048780487805
- gamma           : 1 

---------------------------------------------------------------------------------------------

- name            : forth model 
- split           : 1/5
- oversampling    : 1
- val_acc         : 0.9863945578231292
- test_acc        : 0.9932432432432432
- gamma           : 2

---------------------------------------------------------------------------------------------
- name            : fifth model 
- split           : 1/5
- oversampling    : 0.5
- val_acc         : 0.9863945578231292
- test_acc        : 0.9864864864864865
- gamma           : 2

---------------------------------------------------------------------------------------------



In [47]:
df.columns


Index(['Gender', 'Age', 'Height', 'Weight', 'H_Cal_Consump', 'Veg_Consump',
       'Water_Consump', 'Alcohol_Consump', 'Smoking', 'Meal_Count',
       'Food_Between_Meals', 'Fam_Hist', 'H_Cal_Burn', 'Phys_Act',
       'Time_E_Dev', 'Transport', 'Body_Level'],
      dtype='object')